### ISSUE TITLE
zooming mappings

### ISSUE THREAD

**Issue by albertolg** [opentrack/opentrack#535]
Hi, would be possible to have a bigger resolution for the horizontal value of the axis curve? I mean if is possible to lower the scale value of the X axis from 180 to 50 about so we can have a more precise edit? Right now mine are this, but it's very difficult to obtain a smooth transition, I'm using the edtracker DIY

![1](https://cloud.githubusercontent.com/assets/7831332/22262610/edb41b60-e271-11e6-8f63-ac8f43c3653c.jpg)
![2](https://cloud.githubusercontent.com/assets/7831332/22262611/edb7b8ba-e271-11e6-8546-524ea585107e.jpg)

**Comment by albertolg** [comment]
forget it, modified the scaling to 2x in the edtracker GUI and now is much better

**Comment by albertolg** [comment]
well, I spoke too early... I have too many spikes with 2x so hope this can be implemented thanks

**Comment by Kissamies** [comment]
While an idea worth considering in general sense, can't you use EDTracker UI to crank up the response scaling to get more movement out of your device?

**Comment by LennyPenny** [comment]
@Kissamies the problem is actually the reverse, the edtracker seems to really sensitive which makes editing the curves at this scale so hard

**Comment by albertolg** [comment]
well, I adjusetd the scaling to 2x and worked well, then when I tried later it seemed too jumpy so I reverted back to 1x. I have mapped the roll input to the X axis, I will do more tests with different scaling value

**Comment by sthalik** [comment]
The idea here is that the mapping window wouldn't fit on a smaller display's resolution. We have to take care of the lowest common denominator here.

For users of non-180 deg devices -- pretty much everything non-VR -- we could clamp the mapping graph to the user's chosen value. I have some flu and it's gonna take a few days just to get strong again. I'm also burned out from sole-programmer duties. Can someone do it and submit a pull request please? It's actually really simple.

**Comment by sthalik** [comment]
And to be clear, I'm gonna introduce clamping the mapping to a chosen value. It's been annoying since basically forever. So it's approved.

**Comment by sthalik** [comment]
So test builds are here as usual <<https://www.dropbox.com/sh/823dd4wyhixv5a7/AACB1Iqquvia_keD9jUY2qFBa>>

And the filename is `opentrack-test-20170129_01-zoomed-mappings.7z`

Also please don't call them curves 🐈

**Comment by sthalik** [comment]
Please note that if you have a point that gets cut out by zoom, it won't take effect when zoomed in. But it won't be corrupted or removed Just unzoom to have it.

**Comment by albertolg** [comment]
downloaded and tested, working good... can you please add also for the X Y and Z ? :-)

**Comment by sthalik** [comment]
Is there a point at all to add for position?

**Comment by albertolg** [comment]
well... right now mine is this.. I don't know if is useful or not adding it 
![immagine](https://cloud.githubusercontent.com/assets/7831332/22406362/5f8312d0-e652-11e6-8395-9a89c8bf5d7c.png)

other are working good and easy to edit

![immagine](https://cloud.githubusercontent.com/assets/7831332/22406371/8458c2f8-e652-11e6-80ca-1356b0c568e4.png)

**Comment by petsild** [comment]
Hi,
when I see a new change to work with curves :) I would like to ask if it was possible to create a switch like has graphically now the X-axis do to the Z axis doesn't start at the center of 50% but ranging from 0% - 100% explain why.
In all simulators DCS,ROF,BOS.BOM I'm trying to set the view with the least zoom, and shift in the cockpit for a better overview of what's happening. The wider shot is just clearer and there's more to see.
It would be an improvement for a lot of pilots, thank you for any response. Pet

**Comment by sthalik** [comment]
@petsild I understand. But we're scaling movement along the X axis on the mapping and not along the Y axis. Maybe I can reduce snap when zoomed in. Stay tuned. I also found some tiny tiny bugs in the initial experimental build, so fixing.

**Comment by petsild** [comment]
I thought to make it a manual shift seats in the cockpit using some gate valves. You can now simulate at startup centering to zoom in and then zoom out to the starting position. But when the new centering is the default position back to 50%.

The idea in graphics:
![otzoom](https://cloud.githubusercontent.com/assets/13926997/22595098/5529ddbe-ea26-11e6-864f-cab58b771438.jpg)

**Comment by sthalik** [comment]
@petsild the zooming exists so using only the initial part of the mapping is easier and more precise.

I'll add zooming XYZ.

**Comment by petsild** [comment]
I'm sorry I expressed myself badly only shift the Z-axis in the cockpit.

**Comment by sthalik** [comment]
@petsild can you rephrase your original message please?

**Comment by petsild** [comment]
Idea to shift the centering of the field of view to be sliding freely forward or backward as the seat of the pilot.

**Comment by sthalik** [comment]
But how can you shift it? It's an absolute position taken from raw data.

**Comment by petsild** [comment]
I'm not enough versed in the issue of raw data it will probably be a complication to write the correct interpretation.

**Comment by petsild** [comment]
If it was possible to impose after the centring of the specific games a different position of the Z axis?

**Comment by sthalik** [comment]
@petsild camera -> custom center pose

**Comment by petsild** [comment]
Such an option in the settings of glossed over and didn't know what exactly is intended. Thank you very much and will attend to this.
 Have a nice weekend. Pet

**Comment by petsild** [comment]
Shift using a custom center pose is a very useful feature. Activation happens to the Y-axis is too sensitive and dependent on the axis of the pitch instead of just turning around the center. You can compensate for the blunt curve of it is not so accurate.
![ccp](https://cloud.githubusercontent.com/assets/13926997/22619590/09d902e0-eaf8-11e6-8089-30acd8405705.jpg)

**Comment by sthalik** [comment]
See today's build for zooming mappings implementation. <<https://www.dropbox.com/sh/823dd4wyhixv5a7/AACB1Iqquvia_keD9jUY2qFBa>>.

@petsild the custom center pose isn't mapped, and mappings don't take it into account.

**Comment by albertolg** [comment]
@sthalik changing the scale do nothing to the zoomings X Y and Z

**Comment by sthalik** [comment]
@albertolg fixed in time for 2.3.1. Please report any issues.

**Comment by albertolg** [comment]
seems all ok, great work @sthalik thanks again
